src/dstack/_internal/server/migrations/versions/9be642a33d08_added_user_id_and_user_to_volumes_table.py

```diff
@@ -0,0 +1,45 @@
+"""Added user_id and user to Volumes table
+
+Revision ID: 9be642a33d08
+Revises: afbc600ff2b2
+Create Date: 2024-11-02 01:46:48.989331
+
+"""
+
+import sqlalchemy as sa
+import sqlalchemy_utils
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "9be642a33d08"
+down_revision = "afbc600ff2b2"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table("volumes", schema=None) as batch_op:
+        batch_op.add_column(
+            sa.Column(
+                "user_id", sqlalchemy_utils.types.uuid.UUIDType(binary=False), nullable=False
+            )
+        )
+        batch_op.create_foreign_key(
+            batch_op.f("fk_volumes_user_id_users"),
+            "users",
+            ["user_id"],
+            ["id"],
+            ondelete="CASCADE",
+        )
+
+    # ### end Alembic commands ###

```

Comment:
This migration won't work on existing installations that already have volumes in the db. (You cannot add non-nullable field to existing rows).  I can suggest two solutions:

1. Just make VolumeModel.user_id/VolumeModel.user optional and work around that.
2. Ideally, we'd like to make user required. To do that, you can write a migration that adds a new nullable user field, then populates existing rows with values (e.g. from volume_model.project.owner), then makes the field non-nullable.

Solution 2 is preferable to avoid handling optional user case in the future, but I'm ok with merging Solution 1 as well if you're not feeling like writing data migrations.

Reply:
Good point. let me try and populate existing volumes with the user from volume's project owner.

Reply:
updated migration to set volume.user_id to the projects owner